hack/update-bindata.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o pipefail
+set -x
+
+if [ -z $KUBE_ROOT ]; then
+	echo "KUBE_ROOT not detected, setting default."
+	KUBE_ROOT="../../../"
+fi
+
+if [[ ! -d "${KUBE_ROOT}/examples" ]]; then
+	echo "This is meant to be run from the test/e2e/ directory by test/e2e/e2e.go or similar."
+	exit 1
+fi
+
+echo "running generation framework"
+pwd
+
+# Setup bindata if not already in the system.
+# For separation of concerns, download first, then install later.
+go get -u github.com/jteeuwen/go-bindata/... || echo "go-bindata get failed, possibly already exists, proceeding"
+go install github.com/jteeuwen/go-bindata/ || echo "go-bindata install may have failed, proceeding anyway..."
+
+if [[ ! -f $GOPATH/bin/go-bindata ]]; then 
+	echo "missing bin/go-bindata"
+	echo "for debugging, printing search for bindata files out..."
+	find $GOPATH -name go-bindata
+	exit 5
+fi
+
+BINDATA_OUTPUT="${KUBE_ROOT}/test/e2e/generated/bindata.go"
+$GOPATH/bin/go-bindata -nometadata -prefix "${KUBE_ROOT}" -o ${BINDATA_OUTPUT} -pkg generated \
+	-ignore .jpg -ignore .png -ignore .md \
+	$KUBE_ROOT/examples/* $KUBE_ROOT/docs/user-guide/* $KUBE_ROOT/test/e2e/testing-manifests/* $KUBE_ROOT/test/images/*
```

Comment:
might be better to use `...` instead of globs - as otherwise we may have too many arguments. it might also be worth putting these on separate lines, for the benefit of readability and version control history. in other words,
```shell
    "${KUBE_ROOT}/examples/..." \
    "${KUBE_ROOT}/docs/user-guide/..." \
    "${KUBE_ROOT}/test/e2e/testing-manifests/..." \
    "${KUBE_ROOT}/test/images/..."
```